cbqueryx/query_test.go

```diff
@@ -45,3 +50,73 @@ func TestQuery(t *testing.T) {
 
 	require.Empty(t, cache.queryCache)
 }
+
+func TestQueryIndexExists(t *testing.T) {
+	index := uuid.NewString()[:6]
+	expectedResult := makeErrorQueryResult([]queryErrorJson{
+		{
+			Code: 4300,
+			Msg:  fmt.Sprintf("The index %s already exists.", index),
+			Reason: map[string]interface{}{
+				"name": index,
+			},
+		},
+	})
+	body, err := json.Marshal(expectedResult)
+	require.NoError(t, err)
+
+	resp := &http.Response{
+		Status:        "conflict",
+		StatusCode:    409,
+		Header:        nil,
+		Body:          io.NopCloser(bytes.NewReader(body)),
+		ContentLength: int64(len(body)),
+	}
+
+	opts := &QueryOptions{
+		Statement: fmt.Sprintf("CREATE INDEX %s", index),
+	}
+	_, err = Query{
+		Transport: makeSingleTestRoundTripper(resp, nil),
+		Logger:    testutils.MakeTestLogger(t),
+		UserAgent: "useragent",
+		Username:  "username",
+		Password:  "password",
+	}.Query(context.Background(), opts)
+	assert.ErrorIs(t, err, ErrIndexExists)
+}
+
+func TestQueryIndexNotFound(t *testing.T) {
+	index := uuid.NewString()[:6]
+	expectedResult := makeErrorQueryResult([]queryErrorJson{
+		{
+			Code: 12016,
+			Msg:  fmt.Sprintf("Index Not Found - cause: GSI index %s not found.", index),
+			Reason: map[string]interface{}{
+				"name": index,
+			},
+		},
+	})
+	body, err := json.Marshal(expectedResult)
+	require.NoError(t, err)
+
+	resp := &http.Response{
+		Status:        "success",
+		StatusCode:    200,
```

Comment:
Why does the status and code indicate success if the index could not be found?

Reply:
A good question. Query returns a 200 status code to us in this scenario, I did make sure to verify with wireshark as I asked that question myself after reviewing the gocb code.